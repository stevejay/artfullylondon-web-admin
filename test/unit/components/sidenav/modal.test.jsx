import React from 'react'
import { shallow } from 'enzyme'
import _ from 'lodash'

import SidenavModal from '_src/components/sidenav/modal'

it('should render correctly when showing', () => {
  const wrapper = shallow(
    <SidenavModal show onHide={_.noop}><div id='child' /></SidenavModal>
  )

  expect(wrapper).toMatchSnapshot()
})

it('should render correctly when not showing', () => {
  const wrapper = shallow(
    <SidenavModal show={false} onHide={_.noop}><div id='child' /></SidenavModal>
  )

  expect(wrapper).toMatchSnapshot()
})
