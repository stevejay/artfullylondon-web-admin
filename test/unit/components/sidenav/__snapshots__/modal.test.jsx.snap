// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly when not showing 1`] = `
<Modal
  aria-label="Navigation menu"
  autoFocus={true}
  backdrop={true}
  backdropClassName="backdrop"
  className="src-components-sidenav-modal__modal"
  enforceFocus={true}
  keyboard={true}
  manager={
    ModalManager {
      "add": [Function],
      "containers": Array [],
      "data": Array [],
      "handleContainerOverflow": true,
      "hideSiblingNodes": true,
      "isTopModal": [Function],
      "modals": Array [],
      "remove": [Function],
    }
  }
  onHide={[Function]}
  renderBackdrop={[Function]}
  restoreFocus={true}
  show={false}
  timeout={250}
  transition={[Function]}
>
  <div
    className="nocontent src-components-sidenav-modal__content"
    role="navigation"
  >
    <div
      id="child"
    />
  </div>
</Modal>
`;

exports[`should render correctly when showing 1`] = `
<Modal
  aria-label="Navigation menu"
  autoFocus={true}
  backdrop={true}
  backdropClassName="backdrop"
  className="src-components-sidenav-modal__modal"
  enforceFocus={true}
  keyboard={true}
  manager={
    ModalManager {
      "add": [Function],
      "containers": Array [],
      "data": Array [],
      "handleContainerOverflow": true,
      "hideSiblingNodes": true,
      "isTopModal": [Function],
      "modals": Array [],
      "remove": [Function],
    }
  }
  onHide={[Function]}
  renderBackdrop={[Function]}
  restoreFocus={true}
  show={true}
  timeout={250}
  transition={[Function]}
>
  <div
    className="nocontent src-components-sidenav-modal__content"
    role="navigation"
  >
    <div
      id="child"
    />
  </div>
</Modal>
`;
